controllers/etcd_copy_backups_controller.go

```diff
@@ -0,0 +1,446 @@
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+		if copyTask.Status.LastOperation != nil && copyTask.Status.LastOperation.State == druidv1alpha1.LastOperationStateActive {
+			return ctrl.Result{
+				Requeue: true,
+			}, errors.New("The resource can not be deleted while still processing")
+		}
+		if !reflect.DeepEqual(copyTask.Status.TaskReference.Kind, "") && !reflect.DeepEqual(copyTask.Status.TaskReference.Kind, "Job") {
+			return ctrl.Result{
+				Requeue: true,
+			}, fmt.Errorf("Unsupported task kind %s", copyTask.Status.TaskReference.Kind)
+		}
+
+		job := &batchv1.Job{}
+		if err := r.Get(ctx, kutil.Key(copyTask.Namespace, copyTask.Status.TaskReference.Name), job); err == nil {
+			if err := r.removeFinalizerFromObject(ctx, job, logger); err != nil {
+				return ctrl.Result{
+					Requeue: true,
+				}, err
+			}
+			if err := r.Delete(ctx, job); client.IgnoreNotFound(err) != nil {
+				return ctrl.Result{
+					Requeue: true,
+				}, err
+			}
+		} else if client.IgnoreNotFound(err) != nil {
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+	}
+	if err := r.removeFinalizerFromObject(ctx, copyTask, logger); err != nil {
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	logger.Info(fmt.Sprintf("Deleted ETCDCopyBackupTask %s/%s successfully.", copyTask.Namespace, copyTask.Name))
+	return ctrl.Result{}, nil
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) updateEtcdCopyTaskStatusWithTaskDescription(ctx context.Context, copyTask *druidv1alpha1.EtcdCopyBackupsTask, name, kind, apiGroup string) error {
+	fmt.Printf("updating task status")
+	return kutil.TryUpdateStatus(ctx, retry.DefaultBackoff, r.Client, copyTask, func() error {
+		copyTask.Status.TaskDeployed = true
+		copyTask.Status.TaskReference = &corev1.TypedLocalObjectReference{
+			Name:     name,
+			Kind:     kind,
+			APIGroup: pointer.StringPtr(apiGroup),
+		}
+		return nil
+	})
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) updateEtcdCopyTaskStatusLastError(ctx context.Context, copyTask *druidv1alpha1.EtcdCopyBackupsTask, err error) error {
+	return kutil.TryUpdateStatus(ctx, retry.DefaultBackoff, r.Client, copyTask, func() error {
+		copyTask.Status.LastError = pointer.StringPtr(err.Error())
+		copyTask.Status.ObservedGeneration = &copyTask.Generation
+		return nil
+	})
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) updateEtcdCopyBackupsTaskStatus(ctx context.Context, copyTask *druidv1alpha1.EtcdCopyBackupsTask) error {
+	return kutil.TryUpdateStatus(ctx, retry.DefaultBackoff, r.Client, copyTask, func() error {
+		if copyTask.Status.TaskDeployed && copyTask.Status.TaskReference != nil {
+			lastOperation, err := r.getCopyJobLastOperation(ctx, copyTask)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					if copyTask.Status.TaskDeployed {
+						copyTask.Status.LastError = pointer.StringPtr(fmt.Sprintf("Task %s was forcefully deleted", copyTask.Status.TaskReference.Name))
+					}
```

Comment:
This check is redundant from my perspective, if someone manages to delete the job it's not something we care too much about. We should simply set the reference field to `nil` here as we know it doesn't exist anymore and that's it.